clippy_lints/src/utils/conf.rs

```diff
@@ -205,10 +205,33 @@ macro_rules! define_Conf {
 }
 
 define_Conf! {
-    /// Lint: Arithmetic.
+    /// Lint: arithmetic_side_effects.
     ///
-    /// Suppress checking of the passed type names.
-    (arithmetic_side_effects_allowed: rustc_data_structures::fx::FxHashSet<String> = <_>::default()),
+    /// Suppress checking of the passed type pair names grouped by commas in binary operations like
+    /// addition or multiplication.
+    ///
+    /// Supports the "*" wildcard to indicate that a certain type won't trigger the lint regardless
+    /// of the involved counterpart. For example, `["SomeType", "*"]` or `["*", "AnotherType"]`.
+    ///
+    /// Pairs are asymmetric, which means that `["SomeType", "AnotherType"]` is not the same as
+    /// `["AnotherType", "SomeType"]`.
+    ///
+    /// # Example
+    ///
+    /// ```toml
+    /// arithmetic-side-effects-allowed-bin = [["SomeType" , "f32"], ["AnotherType", "*"]]
+    /// ```
+    (arithmetic_side_effects_allowed_bin: Vec<[String; 2]> = <_>::default()),
```

Comment:
I wouldn't abreviate this
```suggestion
    (arithmetic_side_effects_allowed_binary: Vec<[String; 2]> = <_>::default()),
```

Reply:
Done